From 7f8db8f9cc0cfea9ed7bbb556b2bda1b8a99155a Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Fri, 12 Sep 2014 18:04:58 -0400
Subject: [powerpc] Add more trap names to xmon

Message-id: <1410545655-205645-70-git-send-email-dzickus@redhat.com>
Patchwork-id: 94140
O-Subject: [RHEL7 PATCH 069/626] powerpc: Add more trap names to xmon
Bugzilla: 1127366
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Steve Best <sbest@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: David Gibson <dgibson@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1127366

commit 660e034ce167b0954b83fd024c8be02c2911dbc9
Author: Michael Ellerman <michael@ellerman.id.au>
Date:   Thu Aug 15 15:22:16 2013 +1000

    powerpc: Add more trap names to xmon

    We haven't updated these for a while it seems, it's nice to have in the
    oops output.

    Signed-off-by: Michael Ellerman <michael@ellerman.id.au>
    Signed-off-by: Benjamin Herrenschmidt <benh@kernel.crashing.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/powerpc/xmon/xmon.c b/arch/powerpc/xmon/xmon.c
index 5972100..7696091 100644
--- a/arch/powerpc/xmon/xmon.c
+++ b/arch/powerpc/xmon/xmon.c
@@ -1264,11 +1264,18 @@ const char *getvecname(unsigned long vec)
  case 0x700: ret = "(Program Check)"; break;
  case 0x800: ret = "(FPU Unavailable)"; break;
  case 0x900: ret = "(Decrementer)"; break;
+ case 0x980: ret = "(Hypervisor Decrementer)"; break;
+ case 0xa00: ret = "(Doorbell)"; break;
  case 0xc00: ret = "(System Call)"; break;
  case 0xd00: ret = "(Single Step)"; break;
+ case 0xe40: ret = "(Emulation Assist)"; break;
+ case 0xe60: ret = "(HMI)"; break;
+ case 0xe80: ret = "(Hypervisor Doorbell)"; break;
  case 0xf00: ret = "(Performance Monitor)"; break;
  case 0xf20: ret = "(Altivec Unavailable)"; break;
  case 0x1300: ret = "(Instruction Breakpoint)"; break;
+ case 0x1500: ret = "(Denormalisation)"; break;
+ case 0x1700: ret = "(Altivec Assist)"; break;
  default: ret = "";
  }
  return ret;
-- 
1.7.1